### ISSUE TITLE
Firebase Firestore could not build modules while archiving app 

### ISSUE THREAD

**Issue by jtressle** [firebase/firebase-ios-sdk#1266]
### [REQUIRED] Step 2: Environment

  * Xcode version: 9.3
  * Firebase SDK version: 5.0
  * Firebase Component: Firestore
  * Component version: 0.12.0

### [REQUIRED] Step 3: Describe the problem

I'm unable to finish archiving a project with Firebase 5.0.  There are several closed issues for 4.10 with the same symptoms.

When archiving my project, I get the following errors:

RxLibrary-umbrella.h:23:9: Include of non-modular header inside framework module 'RxLibrary
GRPCCall.h:36:9: Could not build module 'RxLibrary'
ProtoRPC.h:20:9: Could not build module 'GRPCClient'
Firestore.pbrpc.h:21:9: Could not build module 'ProtoRPC'

#### Steps to reproduce:

My pod has the following:
pod 'Firebase/Core'
pod 'Firebase/Firestore'
pod 'Firebase/Auth'
pod 'Firebase/Storage'
pod 'FirebaseUI/Facebook'

Pods output:

Using FBSDKCoreKit (4.33.0)
Using FBSDKLoginKit (4.33.0)
Using Firebase (5.0.0)
Using FirebaseAnalytics (5.0.0)
Using FirebaseAuth (5.0.0)
Using FirebaseCore (5.0.0)
Using FirebaseFirestore (0.12.0)
Using FirebaseInstanceID (3.0.0)
Using FirebaseStorage (3.0.0)
Using FirebaseUI (5.0.0)

#### Relevant Code:

Create project and archive it.  Use SDK 5.0.

**Comment by jtressle** [comment]
I went ahead and updated my pod file to the following:

`pod 'FirebaseCore', :git => 'https://github.com/firebase/firebase-ios-sdk.git', :tag => '5.0.0'
pod 'FirebaseFirestore', :git => 'https://github.com/firebase/firebase-ios-sdk.git', :tag => '5.0.0'
pod 'FirebaseAuth', :git => 'https://github.com/firebase/firebase-ios-sdk.git', :tag => '5.0.0'
pod 'FirebaseStorage', :git => 'https://github.com/firebase/firebase-ios-sdk.git', :tag => '5.0.0'
pod 'FirebaseUI/Facebook'`

I'm still getting the four errors.  

Thanks,

**Comment by paulb777** [comment]
This is https://github.com/grpc/grpc/issues/7406. Also discussed at https://github.com/firebase/firebase-ios-sdk/issues/898#issuecomment-374971715.

The workaround is to comment `#import "transformations/GRXMappingWriter.h"` line from `gRPC_RxLibrary-umbrella.h`.

More discussion at GoogleCloudPlatform/ios-docs-samples#30 and https://github.com/GoogleCloudPlatform/ios-docs-samples/blob/master/speech/Swift/Speech-gRPC-Streaming/BUILDFIXES

**Comment by jtressle** [comment]
Thanks for the workaround. I got everything to work with it.

**Comment by Julianvantveld** [comment]
The workaround paulb777 specified works great on local projects.
However, I'm building using Travis CI. There, it constantly crashes since framework changes are not checked in.

Locally the errors are fixed indeed, but Travis builds still fail.

Thanks in advance!

**Comment by morganchen12** [comment]
Since this is a gRPC issue, the fix will be rolled out in a release of gRPC. It won't be tied to the Firebase release cycle.

**Comment by nadimalam** [comment]
How comes its been so many days and this still hasn't been fixed? Does anyone know if a fix is being worked on for the Pods? I can fix this in a local copy fine, but as @Julianvantveld said this is a huge issue for CI.

**Comment by mmcc007** [comment]
Add the following step to your CICD process after the `pod install` to automate the commenting workaround:
```
sed -i '' s@'#import "transformations\/GRXMappingWriter\.h"'@'\/\/#import "transformations\/GRXMappingWriter\.h'@g Pods/Target\ Support\ Files/gRPC-RxLibrary/gRPC-RxLibrary-umbrella.h
```
Used this a year or so ago to fix a gRPC build. Guess it has not yet been permanently fixed.

**Comment by siejkowski** [comment]
The other CI/CD friendly solution is to add
```
post_install do |installer|
    installer.pods_project.build_configuration_list.build_configurations.each do |configuration|
        configuration.build_settings['CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES'] = 'YES'
    end
end
```

to your `Podfile`. Of course it's just a workaround until the gRPC issue is fixed.

**Comment by wilhuff** [comment]
To get the fix now you need something like this in your Podfile

```
pod 'gRPC', :git => 'https://github.com/grpc/grpc.git', :branch => 'master'
pod 'gRPC-Core', :git => 'https://github.com/grpc/grpc.git', :branch => 'master'
pod 'gRPC-ProtoRPC', :git => 'https://github.com/grpc/grpc.git', :branch => 'master'
pod 'gRPC-RxLibrary', :git => 'https://github.com/grpc/grpc.git', :branch => 'master'
```

Note that I don't recommend this for anything more than playing around.

**Comment by Rovel** [comment]
Hey, thanks for all the info, since I'm on a production build whats the best option?
1. Fallback the pod to the latest stable ex: 5.0.0?
2. Point gRPC to git master?

I know the difference, but I'm not aware of critical bug fixes from 5.2 to 5.0 or the kind of problems I may run if I have to point these gRPC to the git master. any suggestions?

Thanks

**Comment by paulb777** [comment]
@Rovel I recommend the workaround at https://github.com/firebase/firebase-ios-sdk/issues/1266#issuecomment-388561282, since it focused and non-intrusive. Its downside is that it needs to be redone after any `pod install/update`.

**Comment by wilhuff** [comment]
Note that the grpc fix is now available at the v1.13.0-pre1 tag, so you no longer have to target master to pull it in. I still wouldn't recommend this for production use but it's definitely coming.

**Comment by hoantran** [comment]
Thanks for the workaround paulb777 (commented on May 12, 2018)

**Comment by yusuftor** [comment]
When will this be properly integrated into the framework?

**Comment by wilhuff** [comment]
As noted above: the fix is coming from gRPC in the 1.13.0 release. They're now on the third prerelease (v1.13.0-pre3).

**Comment by paulb777** [comment]
gRPC 1.13.0 has now released and I confirmed that my test case successfully archives.
